boards/xtensa/intel_adsp_cavs25/revision.cmake

```diff
@@ -0,0 +1 @@
+# intentionally empty
```

Comment:
I find this very cryptic and the documentation does not seem to mention the possibility of leaving `revision.cmake` empty
https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#custom-revision-cmake-files

Now I plead guilty of ignorance on these topics, but there should be IMHO at least enough comments to give a very vague idea of how this works + a couple specific keywords to starting searching the Zephyr documentation.

Reply:
There's a build warning if this file does not exist, but we don't have any meaningful content to put here. Let me extend the comment a bit. And open to even better ideas.

Reply:
I (finally) read the documentation I'm pretty sure you're supposed to have this in this file:
```
board_check_revision(FORMAT NUMBER)
```
This simply tells the Zephyr build system what revision format you're using, in other words it tells CMake that you are NOT using `intel_adsp_cavs25_A.conf`, `intel_adsp_cavs25_B.conf`, etc.

I'm not sure why it seems to work with an empty file but I would not do that as it's not documented. It could very likely break in the future.

The documentation does not mention the possibility of a "revision-less" board either. It looks like either everything has a revision, or there is a _single_, revision-less board. To be on the safe side and avoid potentially spending hours debugging obscure CMake failures (are CMake issues not obscure ever?), I recommend using a revision always. Then you can go a small extra step and have a nice, consistent and symmetric user interface:
```
west build --board intel_adsp_cavs25@2 # build for 2 cores
west build --board intel_adsp_cavs25@4 # build for 4 cores
```

Reply:
I tried adding a simple board_check_revision() but all variants I tried broke current SOF way of not specifying a revision. Let me try again. There's an option to specify default revision.